internal/gencli/gencli.go

```diff
@@ -354,8 +354,10 @@ func (g *gcli) buildFieldFlags(cmd *Command, msg *descriptor.DescriptorProto, pr
 		if field.OneofIndex != nil {
 			g.buildOneOfSelectors(cmd, msg, prefix)
 
-			in := cmd.InputMessage[strings.LastIndex(cmd.InputMessage, ".")+1:]
-			flags = append(flags, g.buildOneOfFlag(cmd, msg, field, prefix, in != msg.GetName())...)
+			// check if we've recursed into a nested message's oneof
+			isInNested := g.descInfo.Type[cmd.InputMessageType].(*descriptor.DescriptorProto) != msg
```

Comment:
No need to type assert, comparing concrete and interface is OK.

https://play.golang.org/p/FuKSIOZVODU